infrastructure/microvm/providers.go

```diff
@@ -0,0 +1,72 @@
+package microvm
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/afero"
+
+	"github.com/weaveworks-liquidmetal/flintlock/core/ports"
+	"github.com/weaveworks-liquidmetal/flintlock/infrastructure/microvm/cloudhypervisor"
+	"github.com/weaveworks-liquidmetal/flintlock/infrastructure/microvm/firecracker"
+	"github.com/weaveworks-liquidmetal/flintlock/internal/config"
+)
+
+var (
+	errUnknownProvider = errors.New("unknown provider")
+)
+
+// New will create a new instance of a microvm service from the supplied name.
+func New(name string, cfg *config.Config, networkSvc ports.NetworkService, diskSvc ports.DiskService, fs afero.Fs) (ports.MicroVMService, error) {
+	switch name {
+	case firecracker.ProviderName:
+		return firecracker.New(firecrackerConfig(cfg), networkSvc, fs), nil
+	case cloudhypervisor.ProviderName:
+		return cloudhypervisor.New(cloudHypervisorConfig(cfg), networkSvc, diskSvc, fs), nil
+	default:
+		return nil, errUnknownProvider
+
+	}
+}
+
+// NewFromConfig will create instances of the vm providers based on the config.
+func NewFromConfig(cfg *config.Config, networkSvc ports.NetworkService, diskSvc ports.DiskService, fs afero.Fs) (map[string]ports.MicroVMService, error) {
+	providers := map[string]ports.MicroVMService{}
+
+	//TODO: handle this better, disable flags????
```

Comment:
what are you asking here?

Reply:
Comment has been removed. 

I was thinking, do we need explicit flags to disable cloudhypervisor or firecracker? Or is it ok to rely on supplying the path to binary (like we do now)?

Reply:
ah i see. 🤔 i think fine to just have paths now

we may find or get feedback that explicit flags (which maybe also suppose a default bin path?) are a better UX and we can handle that then